src/app/client/src/app/modules/shared/components/player/player.component.ts

```diff
@@ -44,6 +47,13 @@ export class PlayerComponent implements OnInit, OnChanges {
       }
     });
   }
+  /**
+ * To check if the mimeType is text/x-url
+ * if mimeType is text/x-url extcontentpreview plugin will be invoked
+ */
+  checkExtUrl() {
+    this.contentUtilsServiceService.getRedirectUrl(this.playerConfig.metadata, this.userId, this.courseId, this.batchId);
```

Comment:
this component is used from public-collection-player and public-content-player components also. I don't see changes in those components to send user id. 
Two things to consider here:
1. If no user id is passed, what will be the redirect url?
2. For public pages, I guess we should be using "anonymous" as user-id in telemetry. Check with Harish about this and update accordingly.

Reply:
@vrayulu : added 'anonymous' as userid in the public players